src/code/model/link.js

```diff
@@ -0,0 +1,115 @@
+import { generateId } from "../auxiliar";
+import { getSelectedOperation } from "../selectedOp";
+
+export default class WasabeeLink {
+  constructor(obj) {
+    this.ID = obj.ID ? obj.ID : generateId();
+    this.fromPortalId = obj.fromPortalId;
+    this.toPortalId = obj.toPortalId;
+    this.description = obj.description ? obj.description : null;
+    this.assignedTo = obj.assignedTo ? obj.assignedTo : "";
+    this.throwOrderPos = obj.throwOrderPos ? Number(obj.throwOrderPos) : 0;
+    this.color = obj.color ? obj.color : "main";
+    this.completed = obj.completed ? !!obj.completed : false;
+    this.zone = obj.zone ? Number(obj.zone) : 1;
+  }
+
+  // build object to serialize
+  toJSON() {
+    return {
+      ID: this.ID,
+      fromPortalId: this.fromPortalId,
+      toPortalId: this.toPortalId,
+      description: this.description,
+      assignedTo: this.assignedTo,
+      throwOrderPos: Number(this.throwOrderPos),
+      color: this.color,
+      completed: !!this.completed, // !! forces a boolean value
+      zone: Number(this.zone),
+    };
+  }
+
+  // for interface consistency, the other types use comment
+  // we can't rename them here w/o making the corresponding changes on the server
+  get comment() {
+    return this.description;
+  }
+
+  set comment(c) {
+    this.description = c;
+  }
+
+  // for interface consistency, other types use order
+  get opOrder() {
+    return this.throwOrderPos;
+  }
+
+  set opOrder(o) {
+    this.throwOrderPos = Number.parseInt(o, 10);
+  }
+
+  // make the interface match (kinda) what markers do
+  // 'pending','assigned','acknowledged','completed'
+  // THESE ARE INTERNAL VALUES AND SHOULD NOT BE wX'd!!!
+  get state() {
```

Comment:
this will be next work, to have a common base class Task
the data models (from the server) differ a bit (completedID/state vs completed for instance)